src/main/java/ua/okwine/productexpirationdate/controller/ProductController.java

```diff
@@ -0,0 +1,136 @@
+import ua.okwine.productexpirationdate.service.ProductService;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+@Controller
+@RequestMapping("/products")
+public class ProductController {
+
+    private final ProductService productService;
+
+    @Autowired
+    public ProductController(ProductService productService) {
+        this.productService = productService;
+    }
+
+    @GetMapping("/main")
+    public String startPage() {
+        return "startpage";
+    }
+
+    @GetMapping("/newExpDateForm")
+    public String showFormForAdd(Model model) {
+        List<Provider> providers = productService.findAllProviders();
+        model.addAttribute("providers", providers);
+        model.addAttribute("product", new Product());
+
+        return "products/new-expiration-date-form";
+    }
+
+    @GetMapping("/list-products")
+    public String getAll(Model model) {
+        List<Product> productList = productService.findAll();
+
+        model.addAttribute("products", productList);
+
+        return "products/list-products";
+    }
+
+    @GetMapping("/productUploadForm")
+    public String providerUploadForm() {
+        return "products/product-upload-form";
+    }
+
+    @PostMapping("/importProductFromExcel")
+    public String importExcel(Model model, @RequestParam("file") MultipartFile file) {
+        String uploadDirectory = "src/main/resources/temp_files";
+        String fullPath = "src/main/resources/temp_files/" + file.getOriginalFilename();
+
+        Path fileNameAndPath = Paths.get(uploadDirectory, file.getOriginalFilename());
+        try {
+            Files.write(fileNameAndPath, file.getBytes());
+        } catch (IOException e) {
+            e.printStackTrace();
+            model.addAttribute("msg", "Файл " + file.getOriginalFilename() +
+                    " НЕ УДАЛОСЬ ИМПОРТИРОВАТЬ!");
+
+            return "providers/import-result";
+        }
+        model.addAttribute("msg", "Файл " + file.getOriginalFilename() +
+                " успешно загружен.");
+
+        productService.saveFromExcel(fullPath);
+        return "providers/import-result";
+    }
+
+    @PostMapping("/save")
+    public String saveProduct(@ModelAttribute("product") Product product) {
+        productService.save(product);
+
+        return "redirect:/products/newExpDateForm";
+    }
+
+    @GetMapping("/updateProductForm")
+    public String updateProduct(@RequestParam("id") int id,
+                                Model model) {
+        Product product = productService.findById(id);
+        List<Provider> providers = productService.findAllProviders();
+
+        model.addAttribute("product", product);
+        model.addAttribute("providers", providers);
+
+        return "products/new-expiration-date-form";
+    }
+
+    @GetMapping("/settings")
```

Comment:
I don't see how it's related to the product, consider moving it to a separate controller